src/engine/ui/basics/buttonGroup.cxx

```diff
@@ -1,3 +1,70 @@
 #include "buttonGroup.hxx"
 
-void ButtonGroup::addToGroup(UiElement *widget) { m_buttonGroup.push_back(widget); }
+#include "../../uiManager.hxx"
+
+void ButtonGroup::addToGroup(Button *widget) { m_buttonGroup.push_back(widget); }
+
+bool ButtonGroup::onMouseButtonDown(const SDL_Event &event)
+{
+  for (const auto &it : m_buttonGroup)
+  {
+    if (it->isVisible())
+    {
+      if (it->onMouseButtonDown(event))
+      {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+bool ButtonGroup::onMouseButtonUp(const SDL_Event &event)
+{
+  for (const auto &it : m_buttonGroup)
+  {
+    if (it->isMouseOver(event.button.x, event.button.y) && it->isVisible())
+    {
+      if (exclusive)
```

Comment:
`if (it->isMouseOver(event.button.x, event.button.y) && it->isVisible())` and `if (exclusive)` can combine.

Reply:
Thanks! resolved